src/Lykke.Service.HFT.Services/AssetServiceDecorator.cs

```diff
@@ -1,25 +1,30 @@
-﻿using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
+﻿using System;
 using Lykke.Service.Assets.Client;
 using Lykke.Service.Assets.Client.Models;
 using Lykke.Service.HFT.Core.Services;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 
 namespace Lykke.Service.HFT.Services
 {
-    public class AssetServiceDecorator : IAssetServiceDecorator
+    public class AssetServiceDecorator : IAssetServiceDecorator, IDisposable
     {
         private readonly IAssetsServiceWithCache _apiService;
+        private IDisposable _updater;
 
         public AssetServiceDecorator(IAssetsServiceWithCache apiService)
         {
-            _apiService = apiService;
+            _apiService = apiService ?? throw new ArgumentNullException(nameof(apiService));
+            _updater = apiService.StartAutoCacheUpdate();
```

Comment:
I suppose this API is not merged to test branch of AssetsService. We shouldn't use it right now.